### DISCUSSION TITLE
Quarto manuscript cannot properly render tables, supplementary tables, figures, supplementary figures.

### DISCUSSION THREAD

**Discussion by davidzgli** [quarto-dev/quarto-cli#10755] [Q&A] [answered]
### Description

I ran into troubles when rendering to html or typst format document from a quarto file that has tables, supplementary tables, figures, and supplementary figures.  Multiple issues showed up, so I created a github repository to reproduce these issues (https://github.com/davidzgli/manuscript-example).  

When rendering the index.qmd file to html, I get the following error:

1. The cross-references for the mtcars-head table are not rendered properly. The cross-references are displayed as raw strings (e.g. @tbl-mtcars-head), not rendered strings (e.g. Table 1).
2. The same issue occurs for the iris-head supplementary table. The cross-references are displayed as raw strings (e.g. @supptbl-iris-head), not rendered strings (e.g. Supplementary Table 1 ).
3. The embdedded code block for creating the mtcars-summary table is not rendered properly. The tables are not displayed in the html file.
4. The same issue occurs for the iris-summary supplementary table. The tables are not displayed in the html file.
Layout issue. The texts are aligned to the leftmost side of the page. The texts should be aligned with other elements, such as tables and figures.

Some issues were also encountered when rendering the index.qmd file to typst document.  The issues are documented at https://github.com/davidzgli/manuscript-example/blob/main/issues.qmd. 

I am on macOS Sonoma 14.6.1 with quarto 1.5.56.  Thank you for your help!

**Comment by mcanouil** [comment]
@davidzgli Could you try to make your example smaller by removing all pieces not necessary?
Maybe try outside of a Manuscript project.

**Reply by davidzgli** [reply]
Ok, I will try to make a smaller example.  I wanted to demonstrate multiple issues with this example, which make it sort of too long.

**Reply by davidzgli** [reply]
Below is a quarto file. 

=============qmd file starts====================
---

title: "crossref figure panel"
format:
  html: default
  docx: default
  typst: default

---

```{r}
library(tidyverse)
library(patchwork)
```

## Make plots

```{r}
#| label: fig-mtcars-combined

# Scatter plot with color
scatter_color <-
ggplot(mtcars) + 
  geom_point(aes(hp, wt, colour = mpg)) + 
  theme(legend.position = 'bottom')

# Bar plot
bar_plot <- 
ggplot(mtcars) + 
  geom_bar(aes(gear)) + 
  facet_wrap(~cyl) + 
  theme(legend.position = 'bottom')

# Combine the plots
scatter_color + bar_plot + plot_layout(ncol = 2) + plot_annotation(title = 'Combined plots', tag_levels = 'A')
```

@fig-mtcars-combined shows the combined plot of a colored scatter and a bar plot for the `mtcars` dataset. [@fig-mtcars-combined]A indicates weight has positive correlation with hp, with higher weights having larger horsepower, accompanied by lower mpg. We can see that four forward gears are most popular for cars with 4 or 6 cylinders, while 3 gears are most popular for cars with 8 cylinders ([@fig-mtcars-combined] B). 

=============qmd file ends====================

**ISSUE**:  This file can be rendered to html format. When rendering to typst, an error would occur saying "label `<fig-mtcars-combinedA>` does not exist in the document".  If I add a space before the letter "A", e.g., "[@fig-mtcars-combined]A" becomes "[@fig-mtcars-combined] A ", the typst rendering also works.   But I don't need a space to be there.

**Reply by mcanouil** [reply]
Unfortunately, your code is unusable.

Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

**Reply by davidzgli** [reply]
Thank you for the link.  I quoted the qmd file and it looks more readable now.  

````
---
title: "crossref figure panel"
format:
  html: default
  docx: default
  typst: default
---

```{r}
#| label: fig-mtcars-combined

library(tidyverse)
library(patchwork)

# Scatter plot with color
scatter_color <-
ggplot(mtcars) + 
  geom_point(aes(hp, wt, colour = mpg)) + 
  theme(legend.position = 'bottom')

# Bar plot
bar_plot <- 
ggplot(mtcars) + 
  geom_bar(aes(gear)) + 
  facet_wrap(~cyl) + 
  theme(legend.position = 'bottom')

# Combine the plots
scatter_color + bar_plot + plot_layout(ncol = 2) + plot_annotation(title = 'Combined plots', tag_levels = 'A')
```

@fig-mtcars-combined shows the combined plot of a colored scatter and a bar plot for the `mtcars` dataset. [@fig-mtcars-combined]A indicates weight has positive correlation with hp, with higher weights having larger horsepower, accompanied by lower mpg. We can see that four forward gears are most popular for cars with 4 or 6 cylinders, while 3 gears are most popular for cars with 8 cylinders ([@fig-mtcars-combined] B). 

````

Below is the output of `quarto check` command.

```bash
➜  ~ quarto check
Quarto 1.5.56
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.56
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2022.05
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/liz30/Library/TinyTeX/bin/universal-darwin
      Version: 2022

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.4
      Path: /Users/liz30/homebrew/opt/python@3.11/bin/python3.11
      Jupyter: 5.5.0
      Kernels: python3

(|) Checking Jupyter engine render....Traceback (most recent call last):
  File "/Applications/quarto/share/jupyter/jupyter.py", line 21, in <module>
    from notebook import notebook_execute, RestartKernel
  File "/Applications/quarto/share/jupyter/notebook.py", line 15, in <module>
    from yaml import safe_load as parse_string
ModuleNotFoundError: No module named 'yaml'
[✓] Checking Jupyter engine render....OK

```

Let me know in case anything else is needed.

**Reply by mcanouil** [reply]
Sorry, I don't understand.
Are you saying the issue is "only" the space after the reference in Typst?
If so, I assume the following is quite enough:

````qmd
---
title: "crossref figure panel"
format: typst
---

```{r}
#| label: fig-mtcars-combined
plot(1)
```

[@fig-mtcars-combined]A 
````

Also, where did you see the square brackets notation for figure/tables cross-reference?

**Reply by mcanouil** [reply]
There might be a bug here or more likely a Typst limitation, but you could use Quarto feature directly to avoid the issue in the first place and have better cross-reference.
See <https://quarto.org/docs/authoring/cross-references.html#figures>.

````qmd
---
title: "crossref figure panel"
format: typst
---

```{r}
#| label: fig-mtcars-combined
#| fig-cap: "Combined figure"
#| fig-subcap: true
#| layout-ncol: 2
plot(1)
plot(2)
```


@fig-mtcars-combined-1 is the first subfigure.
````

**Reply by mcanouil** [reply, marked as answer]
If you really want to use patchwork and not get proper/cleaner sub-figures cross-reference, you need to use: `` [@fig-mtcars-combined]`\`{=typst}A ``, note the `` `\`{=typst}`` to ensure the reference stops there.

**Reply by cscheid** [reply]
Asking a clarification question upstream: https://github.com/jgm/pandoc/discussions/10180

**Reply by davidzgli** [reply]
Yes, it is much clearer.  I guess I have not yet enough linux skills or pandoc knowledge to push the question to this level of clarity.  When I tried the command `pandoc  --version` on my mac, it bounced back a "command not found" error.

**Comment by davidzgli** [comment]
Yes, this example is enough.  Sorry for making the example too
complicated.   I was planning to use one example to demonstrate multiple
issues.  It turned out to be a bad idea.

**Comment by davidzgli** [comment]
Thank you. It works.  I have many figures in the manuscript I am working
on, and every figure has multiple panels.  The proper cross-reference to
the individual panels is crucial.

**Reply by mcanouil** [reply]
The answer you've selected is definitely not the proper cross-reference, i.e., you need more code, more dependencies and an ugly hack.
The more reliable and cleanest approach is https://github.com/quarto-dev/quarto-cli/discussions/10755#discussioncomment-10628604

**Reply by mcanouil** [reply]
To be clear, whatever you append to the cross-reference won't ever be part of the cross-reference, so the link won't take into account the additional characters you put just after.

**Reply by davidzgli** [reply]
The one I selected is sort of hack, but it appears to be compatible with patchwork.  Patchwork has more flexibilities to arrange the plots in many ways.  I guess I would stick with this "hacky" way for now.   I just cannot make sense what the code "`\`{=typst}" means here.

**Reply by mcanouil** [reply]
Did you look for the `layout` option in Quarto?

**Reply by davidzgli** [reply]
Not fully explore the `layout` option yet.  But I think the Quarto `layout` option is mainly targeted to the page/article layout.  It seems to have limited supports for arrange plots/tables,  such as align axis, collect legends, nest plots, etc.

**Reply by mcanouil** [reply]
I suggest you read the documentation on figures: https://quarto.org/docs/authoring/figures.html

**Reply by davidzgli** [reply]
It seems to have more supports for arranging figures than I thought.  Thank you for the link! I will explore it later.
